### PR TITLE
Add CodeQL pipeline

### PR DESCRIPTION
**Pull request by mdebjit**
Add CodeQL pipeline

### CODE REVIEW

**Comment by mdebjit** [comment]
> @mdebjit please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.
> 
> ```
> @microsoft-github-policy-service agree [company="{your company}"]
> ```
> 
> > Options:
> > 
> > * (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > * (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree company="Microsoft"
> > ```
> 
> Contributor License Agreement

@microsoft-github-policy-service agree company="Microsoft"

File: .github/workflows/codeql.yml [hunk 1]
```diff
@@ -0,0 +1,71 @@
+name: "CodeQL Advanced"
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+  schedule:
+    - cron: '38 10 * * 3'
```
Comment (princepereira) [inline .github/workflows/codeql.yml]: 0 0 * * 0

File: .github/workflows/codeql.yml [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+            ${{ runner.os }}-go-modules-
+
+      # Initialize CodeQL
+      - name: Initialize CodeQL
+        uses: github/codeql-action/init@v3
+        with:
+          languages: ${{ matrix.language }}
+          build-mode: ${{ matrix.build-mode }}
+
+      - name: Verify Dependencies
```
Comment (princepereira) [inline .github/workflows/codeql.yml]: Is this required? it's already part of existing pipeline: https://github.com/microsoft/hnslib/blob/main/.github/workflows/ci.yml

File: .github/workflows/codeql.yml (hunk 1 above)
Comment (mdebjit) [reply .github/workflows/codeql.yml]: Addressed!

File: .github/workflows/codeql.yml (hunk 2 above)
Comment (mdebjit) [reply .github/workflows/codeql.yml]: Removed dependency verification check

**Review by sbangari — APPROVED** [review]

File: .github/workflows/codeql.yml [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+          go-version: '1.22' # Adjust Go version as needed
+
+      # Initialize CodeQL
+      - name: Initialize CodeQL
+        uses: github/codeql-action/init@v3
+        with:
+          languages: ${{ matrix.language }}
+          build-mode: ${{ matrix.build-mode }}
+
+      - name: Build
```
Comment (princepereira) [inline .github/workflows/codeql.yml:52]: This is already part of another workflow. Do we still need it?

**Review by princepereira — APPROVED** [review]

**Review by princepereira — APPROVED** [review]

**Pull request merged**